lib/matplotlib/tests/test_cbook.py

```diff
@@ -591,3 +591,32 @@ def test_warn_external(recwarn):
     cbook._warn_external("oops")
     assert len(recwarn) == 1
     assert recwarn[0].filename == __file__
+
+
+def test_array_patch_perimeters():
+    def check(x, rstride, cstride):
+        rows, cols = x.shape
+        row_inds = list(range(0, rows-1, rstride)) + [rows-1]
+        col_inds = list(range(0, cols-1, cstride)) + [cols-1]
+        polys = []
+        for rs, rs_next in zip(row_inds[:-1], row_inds[1:]):
+            for cs, cs_next in zip(col_inds[:-1], col_inds[1:]):
+                # +1 ensures we share edges between polygons
+                ps = cbook._array_perimeter(x[rs:rs_next+1, cs:cs_next+1]).T
+                polys.append(ps)
+        polys = np.asarray(polys)
+        assert np.array_equal(polys,
+                              cbook._array_patch_perimeters(
+                                  x, rstride=rstride, cstride=cstride))
+
+    def divisors(n):
+        for i in range(1, n // 2 + 1):
```

Comment:
I think you can loop all the way to n+1, it's not as if the efficiency cost was that bad :-)
(also it's a single list comprehension)

Reply:
Yes, I'd just have the one-liner implementation in the helper function instead.